### DISCUSSION TITLE
Feature Request: Save Annotations to external Datasource via Http API / Http Datasource

### DISCUSSION THREAD

**Discussion by KrisztianKaszas** [grafana/grafana#38356] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Being able to save an Annotation from the GUI in an external Datasource using Grafana Http API (-> please add support in simple json datasource as well).

**Why is this needed**:
Upon analysing time series in Grafana specific ranges (anomalies) can be manually detected / specific statuses overridden. Such anomalies should be saved back to the original datasource, where a processing engine (e.g. forecasting engine) can consider such marked events in further actions.

Annotations only living in Grafana reduces their usefullness in post processes after visual data control in Grafana took place.

By simply opening the Annotation API, developers could easily add annotation write support to any datasource.

Thank you, best regards and merry holidays.
Krisztian

**Comment by hugohaggmark** [comment]
Merry holidays @KrisztianKaszas and thank you for this contribution!

**Comment by joeltio** [comment]
Regarding opening up a HTTP API for annotations, this seems to have [already been done as of 4.6](https://grafana.com/docs/grafana/latest/http_api/annotations/).
However, for saving annotations directly from Grafana into the initial datasource, this does not seem to have been done (Related: https://github.com/grafana/grafana/issues/9837).

**Comment by mbehal131** [comment]
I would highly appreciate this feature. Can we expect it to be done in the near future? Is anyone working on it? Thank you.
Best regards,
Martin

**Comment by Remmert-A** [comment]
I also support this feature request! Any news?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
